beginner_source/torchtext_custom_dataset_tutorial.py

```diff
@@ -0,0 +1,383 @@
+#
+# Now, let us consider a batch of source sentences as `X` and a batch of target sentences as `y`.
+# Generally, while training a model, we predict on a batch of `X` and compare the result with `y`.
+# But, a batch in our `dataPipe` is of the form `[(X_1,y_1), (X_2,y_2), (X_3,y_3), (X_4,y_4)]`:
+
+print(list(dataPipe)[0])
+# %%
+# So, we will now convert them into the form: `((X_1,X_2,X_3,X_4), (y_1,y_2,y_3,y_4))`.
+# For this we will write a small function:
+
+def separate_source_target(sequence_pairs):
+    """
+    input of form: `[(X_1,y_1), (X_2,y_2), (X_3,y_3), (X_4,y_4)]`
+    output of form: `((X_1,X_2,X_3,X_4), (y_1,y_2,y_3,y_4))`
+    """
+    sources,targets = zip(*sequence_pairs)
+    return sources,targets
+
+## Apply the function to each element in the iterator
+dataPipe = dataPipe.map(separate_source_target)
+print(list(dataPipe)[0])
+
+# %%
+# Now, we have the data as desired.
+#
+# Padding
+# -------
+# As discussed earlier while building vocabulary, we need to pad shorter sentences in a batch to
+# make all the sequences in a batch of equal length. We can perform padding as follows:
+
+def apply_padding(pair_of_sequences):
+    """
+    Convert sequnces to tensors and apply padding
+    """
+    return (T.ToTensor(0)(list(pair_of_sequences[0])), T.ToTensor(0)(list(pair_of_sequences[1])))
+## `T.ToTensor(0)` returns a transform that converts the sequence to `torch.tensor` and also applies
+# padding. Here, `0` is passed to the constructor to specify the index of the `<pad>` token in the
+# vocabulary.
+dataPipe = dataPipe.map(apply_padding)
+
+# %%
+# Now, we can use the index to string mapping to see how the sequence would look with tokens
+# instead of indices:
+
+sourceItoS = sourceVocab.get_itos()
+targetItoS = targetVocab.get_itos()
+
+def show_some_transformed_sentences(data_pipe):
+    """
+    Function to show how the senetnces look like after applying all transforms.
+    Here we try to print actual words instead of corresponding index
+    """
+    for sources,targets in data_pipe:
+        if sources[0][-1] != 0:
+            continue # Just to visualize padding of shorter sentences
+        for i in range(4):
+            source = ""
+            for token in sources[i]:
+                source += " " + sourceItoS[token]
+            target = ""
+            for token in targets[i]:
+                target += " " + targetItoS[token]
+            print(f"Source: {source}")
+            print(f"Traget: {target}")
+        break
+
+show_some_transformed_sentences(dataPipe)
+# %%
+# In the above output we can observe that the shorter sentences are padded with `<pad>`. Now, we
+# can use this dataPipe while writing our training function.
+#
+# Some parts of this tutorial was inspired from this article:
+# Link: `https://medium.com/@bitdribble/migrate-torchtext-to-the-new-0-9-0-api-1ff1472b5d71\
+# <https://medium.com/@bitdribble/migrate-torchtext-to-the-new-0-9-0-api-1ff1472b5d71>`__.
```

Comment:
nit: directly link the hyperlink to "this article".

Reply:
I have made changes as suggested.